File: Source/Templates/NetCore/GtkSharp.Template.CSharp/content/GtkSharp.Dialog.CSharp/Gtk_Dialog.cs
using System;
using Gtk;
using UI = Gtk.Builder.ObjectAttribute;

namespace GtkNamespace
{
    class Gtk_Dialog : Dialog
    {
        public Gtk_Dialog() : this(new Builder("Gtk_Dialog.glade")) { }

        private Gtk_Dialog(Builder builder) : base(builder.GetObject("Gtk_Dialog").Handle)
        {
            DefaultResponse = ResponseType.Cancel;

            Response += OnResponse;
        }

        private void OnResponse(object o, ResponseArgs args)
        {
            Hide();
        }
    }
}

File: Source/Templates/NetCore/GtkSharp.Template.VBNet/content/GtkSharp.Dialog.VBNet/Gtk_Dialog.vb
Imports System
Imports Gtk
Imports UI = Gtk.Builder.ObjectAttribute

Namespace GtkNamespace
    Public Class Gtk_Dialog
        Inherits Dialog

        Public Sub New (builder as Builder)
            MyBase.New (builder.GetObject("Gtk_Dialog").Handle)

            builder.Autoconnect (Me)
            DefaultResponse = ResponseType.Cancel
            
            AddHandler MyBase.Response, AddressOf Dialog_OnResponse
        End Sub
        
        Public Sub New ()
            Me.New (new Builder ("Gtk_Dialog.glade"))
        End Sub
        
        Private Sub Dialog_OnResponse (ByVal sender As Object, ByVal args As ResponseArgs)
            Hide ()
        End Sub
        
    End Class
End Namespace
